File: brady/meeting_rooms.java
/**
 * Definition of Interval:
 * public class Interval {
 *     int start, end;
 *     Interval(int start, int end) {
 *         this.start = start;
 *         this.end = end;
 *     }
 * }
 */

public class Solution {
    /**
     * @param intervals: an array of meeting time intervals
     * @return: if a person could attend all meetings
     */
  
    // O(nlgn) time, O(1) space
    public boolean canAttendMeetings(List<Interval> intervals) {
        Collections.sort(intervals, (i1, i2) -> {
            return i2.start - i1.start;
        });

        for (int i = 0; i < intervals.size()-1; ++i) {
            if (intervals.get(i).end > intervals.get(i+1).start) return false;
        }

        return true;
    }
}

File: jagraj/21/meetingRooms.py
from typing import List


class Interval(object):
    def __init__(self, start, end):
        self.start = start
        self.end = end


class Solution:
    def can_attend_meetings(self, intervals: List[Interval]) -> bool:
        # Write your code here
        intervals.sort(key=lambda x: x.start)
        for i, k in enumerate(intervals[:-1]):
            if k.end > intervals[i + 1].start:
                return False

        return True


if __name__ == "__main__":
    s = Solution()

    print(s.can_attend_meetings([Interval(0, 30), Interval(5, 10), Interval(15, 20)]))
    print(s.can_attend_meetings([Interval(5, 8), Interval(9, 15)]))
